src/main/java/mate/academy/springboot/swagger/conroller/ProductController.java

```diff
@@ -0,0 +1,119 @@
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<Product, ProductRequestDto, ProductResponseDto> dtoMapper;
+    private final PaginationService paginationService;
+
+    @Operation(summary = "Add product to DB")
+    @PostMapping
+    public ProductResponseDto create(@Parameter(description = "Product to add", required = true,
+            schema = @Schema(implementation = ProductRequestDto.class))
+                                         @RequestBody ProductRequestDto dto) {
+        return dtoMapper.mapToDto(productService.create(dtoMapper.mapToModel(dto)));
+    }
+
+    @Operation(summary = "Get product by id")
+    @GetMapping("/{id}")
+    public ProductResponseDto findById(@PathVariable Long id) {
+        return dtoMapper.mapToDto(productService.findById(id));
+    }
+
+    @Operation(summary = "Delete product by id from DB")
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @Operation(summary = "Update product in DB")
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @Parameter(description = "Product to update",
+            required = true,
+            schema = @Schema(implementation = ProductRequestDto.class))
+                                     @RequestBody ProductRequestDto dto) {
+        Product product = productService.update(id, dtoMapper.mapToModel(dto));
+        return dtoMapper.mapToDto(product);
+    }
+
+    @Operation(summary = "Get all Products by price between two values",
+            description = "It is possible to use pagination")
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> findAllByPriceBetween(@Parameter(
+            description = "A value that indicates the price from "
+                    + "which the limit should be filtered") @RequestParam BigDecimal from,
+                                                          @Parameter(
+                                                                  description = "A value "
+                                                                          + "that indicates "
+                                                                          + "the price to "
+                                                                          + "which the "
+                                                                          + "limit should "
+                                                                          + "be filtered")
+                                                          @RequestParam BigDecimal to,
+                                                          @Parameter(
+                                                                  description = "Number "
+                                                                          + "of products per page.")
```

Comment:
```suggestion
                                                                          + "of products per page.", required = false)
```
as well, where the fields are optional